query-algebrizer/src/cc.rs

```diff
@@ -334,10 +334,25 @@ impl ConjoiningClauses {
         self.value_bindings.get(var).cloned()
     }
 
-    pub fn bind_column_to_var(&mut self, table: TableAlias, column: DatomsColumn, var: Variable) {
+    pub fn bind_column_to_var(&mut self, schema: &Schema, table: TableAlias, column: DatomsColumn, var: Variable) {
         // Do we have an external binding for this?
         if let Some(bound_val) = self.bound_value(&var) {
             // Great! Use that instead.
+            // We expect callers to do things like bind keywords here; we need to translate these
+            // before they hit our constraints.
+            // TODO: recognize when the valueType might be a ref and also translate entids there.
+            if column != DatomsColumn::Value {
+                if let TypedValue::Keyword(ref kw) = bound_val {
+                    if let Some(entid) = self.entid_for_ident(schema, kw) {
+                        self.constrain_column_to_entity(table, column, entid);
+                    } else {
+                        // Impossible.
+                        self.mark_known_empty(EmptyBecause::UnresolvedIdent(kw.clone()));
+                    }
+                    return;
+                }
```

Comment:
Shouldn't there be some additional `mark_known_empty` based on non-`Keyword` bindings?  That is, if I give `[$a ...]` and `$a` is bound to `"string value"`, we can reject immediately.  Or does this happen elsewhere as well?